application/server/configuration.go

```diff
@@ -344,10 +344,22 @@ func updateSnykCodeQuality(settings lsp.Settings) {
 }
 
 // TODO store in config, move parsing to CLI
-func updatePath(settings lsp.Settings) {
-	err := os.Setenv("PATH", cachedOriginalPath+string(os.PathListSeparator)+settings.Path)
-	if err != nil {
-		log.Err(err).Msgf("couldn't add path %s", settings.Path)
+func updatePathFromSettings(settings lsp.Settings) {
+	// when changing the path from settings, we cache the original path first, to be able to restore it later
+	if len(cachedOriginalPath) == 0 {
```

Comment:
I'd find it easier to check against `""` instead of using length. But just a preference (and maybe less instructions for the CPU, not that it's important here).